src/ducks/Signals/signalFormManager/signalCrudForm/formParts/ProjectsSelector/TriggerProjectsSelector.js

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useEffect } from 'react'
+import Dialog from '@santiment-network/ui/Dialog'
+import Icon from '@santiment-network/ui/Icon'
+import SearchProjects from '../../../../../../components/Search/SearchProjects'
+import { hasAssetById } from '../../../../../../components/WatchlistPopup/WatchlistsPopup'
+import ProjectsList from './ProjectsList'
+import styles from './TriggerProjectsSelector.module.scss'
+
+export const TriggerProjectsSelector = ({
+  projects = [],
+  setFieldValue,
+  onChange,
+  fieldValueList,
+  values: { target = [] },
+  name
+}) => {
+  const [isOpen, setOpen] = useState(false)
+
+  const [listItems, setListItems] = useState([])
+
+  const toggleAsset = ({ project, listItems, isAssetInList }) => {
+    selected(
+      isAssetInList
+        ? listItems.filter(({ id }) => id !== project.id)
+        : [...listItems, project]
+    )
+  }
+
+  const selected = selectedAssets => {
+    listItems.length !== selectedAssets.length && setListItems(selectedAssets)
+    setFieldValue(name, selectedAssets)
+    onChange && onChange(selectedAssets)
+  }
+
+  useEffect(
+    () => {
+      if (!fieldValueList || fieldValueList.length === 0) {
+        const targetAssets = Array.isArray(target) ? target : [target]
+
+        if (targetAssets.length > 0 && projects.length > 0) {
+          const preSelected = projects.filter(item => {
+            return targetAssets.some(t => t.value === item.slug)
+          })
```

Comment:
Only `slug` property is used on the `item` object, so it should be destructured , otherwise it's the same property access on the every iteration of the `some` function (`LOC 42`).
Shortened code:
```js
projects.filter(({slug}) => targetAssets.some(({value}) => value === slug))
```